sdl-core/src/main/scala/io/smartdatalake/config/ConfigLoader.scala

```diff
@@ -188,7 +196,7 @@ object ConfigLoader extends SmartDataLakeLogger {
             .partition(_.isDirectory)
           (files ++ directories).flatMap(p => getFilesInBfsOrder(p.getPath)).toSeq
       }
-    } else if (fs.isFile(path) && ConfigFile.canHandleExtension(path)) {
+    } else if (fs.getFileStatus(path).isInstanceOf[FileStatus] && ConfigFile.canHandleExtension(path)) {
```

Comment:
Type checks should only be used in rare exceptions in Scala, better:
```suggestion
    } else if (fs.getFileStatus(path).isFile && ConfigFile.canHandleExtension(path)) {
```